### ISSUE TITLE
[mono]pack nuget packages with -cmakeargs orrcured error on windows.

### ISSUE THREAD

**Issue by srxqds** [dotnet/runtime#76717]
### Description

when I build to generate package with -cmakeargs, it hit error:
> ./build.cmd -pack -runtimeFlavor mono /p:Deterministic=true -cmakeargs -DENABLE_OVERRIDABLE_ALLOCATORS=1  -c Release -v d

but it can build successful using:
> ./build.cmd mono+libs -c Release

>  ./build.cmd -pack -runtimeFlavor mono /p:Deterministic=true -cmakeargs -DENABLE_OVERRIDABLE_ALLOCATORS=1
### Reproduction Steps

./build.cmd -pack -runtimeFlavor mono /p:Deterministic=true -cmakeargs -DENABLE_OVERRIDABLE_ALLOCATORS=1  -c Release

### Expected behavior

build success

### Actual behavior

hit error.

**Comment by srxqds** [comment]
@akoeplinger could you help me?

**Comment by srxqds** [comment]
![image](https://user-images.githubusercontent.com/5187024/194404358-4f3b6c6c-ced4-4fd6-9716-4ee6187616f0.png)

the problem is why it was built for linux.x64.release?

**Comment by srxqds** [comment]
should i install zlib on windows which not found in requirements

**Comment by akoeplinger** [comment]
Hmm it looks like there's some issue with passing this to the coreclr build?
If you want to just experiment then you can add the define here instead: https://github.com/dotnet/runtime/blob/eb02b202c281188e9187d3c292cc223107d35a25/src/mono/mono.proj#L217

**Comment by srxqds** [comment]
yeah, it can be fixed by adding define in CMakeList.txt or mono.proj file directly.
I think we the problem is introduced at this line:
https://github.com/dotnet/runtime/blob/415a41770bdf8efd4c3217e2c281233ee5cd03ea/src/coreclr/build-runtime.cmd#L356

we should append __ExtraCmakeArgs with `"-DENABLE_OVERRIDABLE_ALLOCATORS=1" "-DCLR_CROSS_COMPONENTS_BUILD=1"` instead of `"-DENABLE_OVERRIDABLE_ALLOCATORS=1 -DCLR_CROSS_COMPONENTS_BUILD=1"` 

I have tryed but not work
